### PR TITLE
+str #18099 FAILED: TcpSpec

### PR DESCRIPTION
**Pull request by agolubev**
there was a race condition between two connections. From time to time second connection that supposed to be rejected connects before the first one.

### CODE REVIEW

**Comment by agolubev** [comment]
Ref #18099

**Comment by 2m** [comment]
OK TO TEST

File: akka-stream-tests/src/test/scala/akka/stream/io/TcpSpec.scala [hunk 1]
```diff
@@ -503,7 +503,11 @@ class TcpSpec extends AkkaSpec("akka.io.tcp.windows-connection-abort-workaround-
 
     "shut down properly even if some accepted connection Flows have not been subscribed to" in assertAllStagesStopped {
       val address = temporaryServerAddress()
-      val takeTwoAndDropSecond = Flow[IncomingConnection].grouped(2).take(1).map(_.head)
+      val firstClientConnected = Promise[Unit]()
+      val takeTwoAndDropSecond = Flow[IncomingConnection].map(conn ⇒ {
+        if (!firstClientConnected.isCompleted) firstClientConnected.success()
```
Comment (drewhk) [inline akka-stream-tests/src/test/scala/akka/stream/io/TcpSpec.scala]: no need for the if, just use trySuccess()

**Comment by drewhk** [comment]
LGTM

**Comment by drewhk** [comment]
Test failure is weird...

File: akka-stream-tests/src/test/scala/akka/stream/io/TcpSpec.scala (hunk 1 above)
Comment (agolubev) [inline akka-stream-tests/src/test/scala/akka/stream/io/TcpSpec.scala]: will change

**Comment by agolubev** [comment]
Test "Recipe for a droppy broadcast.must work" failed. I think it does not "work" :)

**Comment by drewhk** [comment]
LGTM, Thanks again!

**Pull request merged**